### ISSUE TITLE
[Bug]: MTP produces different tokens than non-MTP at temperature=0 in eager mode

### ISSUE THREAD

**Issue by yangsiqt** [vllm-project/vllm#42513]
### Your current environment

==============================
        System Info
==============================
OS                           : Ubuntu 22.04.1 LTS (x86_64)
GCC version                  : (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Libc version                 : glibc-2.35

==============================
       PyTorch Info
==============================
PyTorch version              : 2.10.0+cu128
Is debug build               : False
CUDA used to build PyTorch   : 12.8

==============================
      Python Environment
==============================
Python version               : 3.10.8

==============================
       CUDA / GPU Info
==============================
GPU models and configuration : NVIDIA GeForce RTX 3090 (24GB)
Nvidia driver version        : 580.76.05
CUDA Version                 : 13.0

==============================
Versions of relevant libraries
==============================
flashinfer-python            : 0.6.6
nvidia-cublas-cu12           : 12.8.4.1
nvidia-cuda-runtime-cu12     : 12.8.90
nvidia-cudnn-cu12            : 9.10.2.21
torch                        : 2.10.0
transformers                 : 4.57.6
triton                       : 3.6.0

==============================
         vLLM Info
==============================
vLLM Version                 : 0.19.1
CUDA Archs                   : Not Set; ROCm: Disabled
Attention Backend            : FLASH_ATTN (FlashAttention v2)

### 🐛 Describe the bug

MTP n=1 produces different token sequences than non-MTP decoding at temperature=0 when enforce_eager=True. The first mismatch occurs at a consistent, reproducible position (e.g., pos 99 for "Hello, my name is"). With enforce_eager=False (CUDA graph enabled), outputs are 100% identical.

This is a numerical determinism bug. The root cause chain:

MTP verification forward uses batch_size=2 (original token + draft token), non-MTP decode uses batch_size=1
In eager mode, cuBLAS auto-tuner selects different GEMM algorithms for different batch sizes, with different internal accumulation orders
Attention output differs by 1–2 ULP (BF16) between the two paths
This ULP error is written into the KV cache and amplified through the attention loop (each step reads all historical K/V)
After ~50–100 steps, accumulated error crosses the logit decision boundary → argmax flips
CUDA graph (enforce_eager=False) fixes this by locking kernel selection during warmup, ensuring identical kernel launch at every step.

This is distinct from PR #38938 (which fixed batch invariance for lm_head and RMSNorm under EAGLE). The attention GEMM itself is not covered by existing batch invariance overrides.

### Before submitting a new issue...

- [ ] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
